src/realm/sync/client.hpp

```diff
@@ -316,7 +316,11 @@ class Session {
         // Will be called after a download message is received and validated by
         // the client but befefore it's been transformed or applied. To be used in
         // testing only.
-        std::function<void()> on_download_message_received_hook;
+        std::function<void(const sync::SyncProgress&, int64_t, sync::DownloadBatchState)>
+            on_download_message_received_hook;
+        /// Will be called after each download message is integrated. For use in testing.
```

Comment:
nit: this is a bit misleading since we do not guarantee this is called for each download messages received (i.e., we "construct" our own download messages through peek_pending)

Reply:
I've copied the comment in src/realm/sync/config.hpp for this parameter here since I think that one is actually accurate.